error/s2n_errno.c

```diff
@@ -313,6 +313,7 @@ static const char *no_such_error = "Internal s2n error";
     ERR_ENTRY(S2N_ERR_INVALID_SERIALIZED_CONNECTION, "Serialized connection is invalid"); \
     ERR_ENTRY(S2N_ERR_TOO_MANY_CAS, "Too many certificate authorities in trust store"); \
     ERR_ENTRY(S2N_ERR_BAD_HEX, "Could not parse malformed hex string"); \
+    ERR_ENTRY(S2N_ERR_NULL_CONFIG_SIGNAL, "Used for testing purposes only. DO NOT throw this error in general."); \
     /* clang-format on */
```

Comment:
Idk about this error message. Maybe just make the usage clearer so nobody wants to use it, and maybe put a comment on it? Like maybe S2N_ERR_CONFIG_NULL_BEFORE_CH_CALLBACK / "Config set to NULL before client hello callback. This should not be possible outside of tests."

You could also add a simple mistakes grep for it.

Reply:
I changed the error name and corresponding string.